sdks/go/sdk.go

```diff
@@ -136,17 +145,21 @@ func (s *SDK) WatchGameServer(f GameServerCallback) error {
 	go func() {
 		for {
 			var gs *sdk.GameServer
+			// Receive the next GameServer event from the stream.
 			gs, err = stream.Recv()
 			if err != nil {
 				if err == io.EOF {
-					_, _ = fmt.Fprintln(os.Stderr, "gameserver event stream EOF received")
 					return
 				}
-				_, _ = fmt.Fprintf(os.Stderr, "error watching GameServer: %s\n", err.Error())
 				// This is to wait for the reconnection, and not peg the CPU at 100%.
 				time.Sleep(time.Second)
 				continue
 			}
+
+			if !reflect.ValueOf(gs.ObjectMeta.DeletionTimestamp).IsZero() {
```

Comment:
Two things:

1. https://pkg.go.dev/time#Time.IsZero exists - no need to reflect.
2. We still need to have at least similar log output as before - right now we're reversing what is required.

My suggestion on a function previously could even be within `WatchGameServer`, something like:

```go
var log = function(gs *GameServer, msg string, err error) {
  if gs.ObjectMeta.DeletionTimestamp.IsZero() {
    return
  }
  ErrorLog(err, msg)
}
```
Then each of the log statements to stderr could become:

```go
log(gs, nil, "gameserver event stream EOF received")
```

and 

```go
logf(gs, err, "error watching GameServer")
```

respectively.

make sense?

Reply:
1. Removed reflect and added `time.Unix(gs.ObjectMeta.DeletionTimestamp, 0).IsZero()`
2. The code defines a function called 'log' that checks if the 'DeletionTimestamp' is zero. If it is, the function returns nothing, otherwise the function calls the 'Logger' function with a message and an error parameter.
       Included log statement for the condition "err == io.EOF" and other errors.